src/main/java/io/jenkins/plugins/checks/github/CheckRunGHEventSubscriber.java

```diff
@@ -97,6 +97,12 @@ private void scheduleRerun(final GHEventPayload.CheckRun checkRun, final String
             if (source.isPresent() && source.get().getRepoOwner().equals(repository.getOwnerName())
                     && source.get().getRepository().equals(repository.getName())
                     && job.getName().equals(branchName)) {
+                if (!gitHubSCMFacade.findHeadCommit(source.get(), job.getLastBuild())
+                        .equals(checkRun.getCheckRun().getHeadSha())) {
```

Comment:
I don’t think we want this tbh, I would just rerun the last build otherwise it will likely confuse users